tools/autograd/gen_variable_type.py

```diff
@@ -658,6 +662,18 @@ def emit_any_requires_grad() -> List[str]:
         return [SETUP_ANY_REQUIRES_GRAD.substitute(
             args_with_derivatives=[arg.name for arg in args_with_derivatives]), ]
 
+    def emit_assert_no_inference_tensor() -> List[str]:
+        tensor_args = []
+        for arg in f.func.arguments.non_out:
+            if isinstance(arg, TensorOptionsArguments):
```

Comment:
Why is this one skipped separately? The tensor_list check below should be enough no?

Reply:
Oh it's because `TensorOptionsArguments` doesn't have `argument` attribute. so we need handle it specially anyway.